.pre-commit-config.yaml

```diff
@@ -1,6 +1,6 @@
 repos:
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.9.6
+    rev: v0.12.2
     hooks:
       - id: ruff
         args: [--fix, --exit-non-zero-on-fix]
```

Comment:
let's also make it so that pre-commit.ci doesn't push Ruff autofixes to our branches, as per https://github.com/python/typing_extensions/pull/628#issuecomment-3046623599

```suggestion
```